test/integration/marked-spec.js

```diff
@@ -1,17 +1,453 @@
-var marked = require('../../marked.min.js');
+var marked = require('../../lib/marked.js');
+var cmSpec = require('./commonmark.json');
+var HtmlDiffer = require('html-differ').HtmlDiffer,
+    htmlDiffer = new HtmlDiffer();
+var since = require('jasmine2-custom-message');
 
-it('should run the test', function () {
-  expect(marked('Hello World!')).toBe('<p>Hello World!</p>\n');
+var Messenger = function() {}
+
+Messenger.prototype.message = function(spec, expected, actual) {
+  return 'CommonMark (' + spec.section + '):\n' + spec.markdown + '\n------\n\nExpected:\n' + expected + '\n------\n\nMarked:\n' + actual;
+}
+
+Messenger.prototype.test = function(spec, section, ignore) {
+  if (spec.section === section && ignore.indexOf(spec.example) < 0) {
+    var expected = spec.html;
+
+    var actual = marked(spec.markdown, { headerIds: false, xhtml: true });
```

Comment:
the `actual` and `expected` variables should be assigned inside the `it` function otherwise they will be calculated before any `beforeEach/All` functions